dpop/DPoP-client-app/src/main/java/org/wso2/dpop/client/DPOPProofGenerator.java

```diff
@@ -0,0 +1,98 @@
+package org.wso2.dpop.client;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JOSEObjectType;
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.crypto.ECDSASigner;
+import com.nimbusds.jose.crypto.RSASSASigner;
+import com.nimbusds.jose.jwk.Curve;
+import com.nimbusds.jose.jwk.ECKey;
+import com.nimbusds.jose.jwk.JWK;
+import com.nimbusds.jose.jwk.RSAKey;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+
+import java.security.InvalidAlgorithmParameterException;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.interfaces.ECPublicKey;
+import java.security.interfaces.RSAPublicKey;
+import java.text.ParseException;
+import java.util.Date;
+import java.util.Scanner;
+import java.util.UUID;
+
+import static com.nimbusds.jose.JWSAlgorithm.ES256;
+import static com.nimbusds.jose.JWSAlgorithm.RS384;
+
+public class DPOPProofGenerator {
+
+    public static void main(String args[]) throws NoSuchAlgorithmException, JOSEException, ParseException,
+            InvalidAlgorithmParameterException {
+
+        Scanner scan = new Scanner(System.in);
+        System.out.println("Enter the public and private key pair type(EC or RSA): ");
+        String keyPairType = scan.nextLine();
+        System.out.println("Enter the HTTP Method: ");
+        String httpMethod = scan.nextLine();
+        System.out.println("Enter the HTTP Url: ");
+        String httpUrl = scan.nextLine();
+
+        KeyPairGenerator gen = KeyPairGenerator.getInstance(keyPairType);
+        KeyPair keyPair;
+        JWK jwk = null;
+        if (keyPairType.equals("EC")) {
```

Comment:
Shall we change the if condition as `"EC".equals(keypairType)` to avoid NPEs?

Reply:
Done